Assets/Shopify/UIToolkit/ShopControllers/MultiProductShopController.cs

```diff
@@ -5,22 +5,33 @@
     using UnityEngine;
 
     public class MultiProductShopController : ShopControllerBase {
-        public new IMultiProductShop Shop { 
-            get { 
-                return base.Shop as IMultiProductShop; 
+        private string cursor = null;
+        private bool allProductsLoaded = false;
+
+        public new IMultiProductShop Shop {
+            get {
+                return base.Shop as IMultiProductShop;
             }
 
             set {
                 base.Shop = value;
-            } 
+            }
         }
 
         public override void OnHide() {
             throw new System.NotImplementedException();
         }
 
         public override void OnShow() {
-            Client.products(OnProductsLoaded);
+            LoadMoreProducts ();
+        }
+
+        public void LoadMoreProducts() {
+            if (allProductsLoaded) {
+                return;
+            }
+
+            Client.products(OnProductsLoaded, after: cursor);
```

Comment:
hmm, wondering if the theme should be sending the cursor? or the number of products to load? In our case we dont really care but imagine an app that has a more classic pagination style, they would want to specify what products to load right?